.github/workflows/build_and_test_cmake.yaml

```diff
@@ -53,12 +53,18 @@ jobs:
       with:
         submodules: true
         ref: ${{ github.event.pull_request.head.sha }}
+        fetch-depth: ${{ !contains(github.event.pull_request.labels.*.name, 'rebase') && 1 || 0 }}
+    - name: Rebase PR
+      if: ${{ github.event.pull_request.head.sha != '' && contains(github.event.pull_request.labels.*.name, 'rebase') }}
+      shell: bash
```

Comment:
Is re-basing during PR supposed to protect us from merging code which was not tested after rebase to latest main branch ? It won't help if all tests pass but user will decide to press the merge button after several days.

We could enforce `--ff-only` merges. Not sure if github supports it out of the box

Reply:
Idea is that it will lover chances of merging such code. We may implement on: push checks later